VSTS/build.ps1

```diff
@@ -1,14 +1,118 @@
-# Download and extract the specified version of the SonarSource Scanner for MSBuild
-$sonar_scanner_version = "2.2"
-$sonar_scanner_full_version = "$sonar_scanner_version.0.24"
-$sonar_scanner_url = "https://github.com/SonarSource-VisualStudio/sonar-scanner-msbuild/releases/download/$sonar_scanner_version/sonar-scanner-msbuild-$sonar_scanner_full_version.zip"
-$sonar_scanner_msbuild_file_name = "$PSScriptRoot\\sonar-scanner-msbuild.zip"
+# ----------------------------------------------------------------------
+# --- Variables
+# ----------------------------------------------------------------------
+# Formatting
+$sectionDisplayFormat = (("-"*25) + "[{0}]" + ("-"*25))
+# Directories
+$buildOutputDir = "$PSScriptRoot\.buildOutput"
+# Visual Studio MarketPlace
+$publisherName = "amaurysonarsource"
+$extensionId = "sonarqube-scanner-msbuild"
+$extensionVersion = "0.2.0"
 
-(New-Object System.Net.WebClient).DownloadFile($sonar_scanner_url, $sonar_scanner_msbuild_file_name)
+# Clean
+function Clean
+{
+	Write-Output ($sectionDisplayFormat -f "Clean")
 
-Add-Type -AssemblyName "System.IO.Compression.FileSystem"
-[IO.Compression.ZipFile]::ExtractToDirectory($sonar_scanner_msbuild_file_name, "$PSScriptRoot\\Tasks\\SonarQubeScannerMsBuildBegin\\SonarQube.Bootstrapper")
+	If (Test-Path $buildOutputDir) 
+	{
+		Write-Output "Deleting folder" $buildOutputDir
+		Remove-Item $buildOutputDir -recurse
+	}
+}
 
+function PatchExtensionVersion
+{
+	param(
+		[Parameter(Mandatory=$True)]
+		[string]$jsonFile
+	)
+	
+	$content = Get-Content $jsonFile -raw | ConvertFrom-Json
+	$content.version = $extensionVersion
+	$content | ConvertTo-Json | Set-Content $jsonFile
+}
 
+function PatchTaskVersion
+{
+	param(
+		[Parameter(Mandatory=$True)]
+		[string]$jsonFile
+	)
+	
+	$splittedVersion = $extensionVersion -split "."

```

Comment:
You could add a verification here to make sure you don't put in a bogus version string